test/GoodGhostingPolygonTest.js

```diff
@@ -0,0 +1,708 @@
+            await truffleAssert.reverts(GoodGhostingPolygon.new(
+                token.address,
+                pap.address,
+                segmentCount,
+                segmentLength,
+                segmentPayment,
+                15,
+                adminFee,
+                pap.address,
+                merkleRoot,
+                incentiveController.address,
+                incentiveController.address,
+                { from: admin },
+            ));
+        });
+
+        it("reverts if the contract is deployed with admin fee more than 20%", async () => {
+            pap = await LendingPoolAddressesProviderMock.new("TOKEN_NAME", "TOKEN_SYMBOL", { from: admin });
+            aToken = await IERC20.at(await pap.getLendingPool.call());
+            await pap.setUnderlyingAssetAddress(token.address);
+            await truffleAssert.reverts(GoodGhostingPolygon.new(
+                token.address,
+                pap.address,
+                segmentCount,
+                segmentLength,
+                segmentPayment,
+                fee,
+                30,
+                pap.address,
+                merkleRoot,
+                incentiveController.address,
+                incentiveController.address,
+                { from: admin },
+            ));
+        });
+    });
+
+    describe("when the contract is deployed", async () => {
+        it("checks if the contract's variables were properly initialized", async () => {
+            const inboundCurrencyResult = await goodGhosting.daiToken.call();
+            const interestCurrencyResult = await goodGhosting.adaiToken.call();
+            const lendingPoolAddressProviderResult = await goodGhosting.lendingPoolAddressProvider.call();
+            const lastSegmentResult = await goodGhosting.lastSegment.call();
+            const segmentLengthResult = await goodGhosting.segmentLength.call();
+            const segmentPaymentResult = await goodGhosting.segmentPayment.call();
+            assert(inboundCurrencyResult === token.address, `Inbound currency doesn't match. expected ${token.address}; got ${inboundCurrencyResult}`);
+            assert(interestCurrencyResult === aToken.address, `Interest currency doesn't match. expected ${aToken.address}; got ${interestCurrencyResult}`);
+            assert(lendingPoolAddressProviderResult === pap.address, `LendingPoolAddressesProvider doesn't match. expected ${pap.address}; got ${lendingPoolAddressProviderResult}`);
+            assert(new BN(lastSegmentResult).eq(new BN(segmentCount)), `LastSegment info doesn't match. expected ${segmentCount}; got ${lastSegmentResult}`);
+            assert(new BN(segmentLengthResult).eq(new BN(segmentLength)), `SegmentLength doesn't match. expected ${segmentLength}; got ${segmentLengthResult}`);
+            assert(new BN(segmentPaymentResult).eq(new BN(segmentPayment)), `SegmentPayment doesn't match. expected ${segmentPayment}; got ${segmentPaymentResult}`);
```

Comment:
shouldn't we check for `incentiveController` and `dataProvider` params passed to the contract?

Reply:
will add for incentiveController dataProvider is not stored in contract just used in constructor in memory